checker/src/main/java/org/checkerframework/checker/regex/RegexTransfer.java

```diff
@@ -79,69 +79,23 @@ public RegexTransfer(CFAbstractAnalysis<CFValue, CFStore, CFTransfer> analysis)
             ExecutableElement method,
             TransferResult<CFValue, CFStore> result) {
         RegexAnnotatedTypeFactory factory = (RegexAnnotatedTypeFactory) analysis.getTypeFactory();
+
+        // Special cases for some RegexUtil.isRegex and RegexUtil.asRegex methods.  Each one adds an
+        // annotation with elements/arguments if possible, or falls back to just `@Regex`.
+        // (No special case is needed for isRegex(String) or asRegex(String); their annotations are
+        // sufficient.)
+
         if (ElementUtils.matchesElement(method, IS_REGEX_METHOD_NAME, String.class, int.class)) {
-            // RegexUtil.isRegex(s, groups) method
-            // (No special case is needed for isRegex(String) because of
-            // the annotation on that method's definition.)
-
-            CFStore thenStore = result.getRegularStore();
-            CFStore elseStore = thenStore.copy();
-            ConditionalTransferResult<CFValue, CFStore> newResult =
-                    new ConditionalTransferResult<>(result.getResultValue(), thenStore, elseStore);
-            JavaExpression firstParam = JavaExpression.fromNode(n.getArgument(0));
-
-            // add annotation with correct group count (if possible,
-            // regex annotation without count otherwise)
-            Node count = n.getArgument(1);
-            int groupCount;
-            if (count instanceof IntegerLiteralNode) {
-                IntegerLiteralNode iln = (IntegerLiteralNode) count;
-                groupCount = iln.getValue();
-            } else {
-                groupCount = 0;
-            }
-            AnnotationMirror enhancedRegexAnnotation =
-                    factory.createEnhancedRegexAnnotation(
-                            new ArrayList<>(Arrays.asList(0, groupCount)));
-            if (thenStore.getValue(firstParam) != null) thenStore.clearValue(firstParam);
-            thenStore.insertValue(firstParam, enhancedRegexAnnotation);
-            return newResult;
+            return getIsRegexModifiedStore(n, result, factory);
```

Comment:
I don't think that you need to abstract out the method here.  Instead of 

```java
        if (ElementUtils.matchesElement(method, IS_REGEX_METHOD_NAME, String.class, int.class)) {
            return getIsRegexModifiedStore(n, result, factory);
        } else if (ElementUtils.matchesElement(
                method, IS_REGEX_METHOD_NAME, String.class, int.class, int[].class)) {
            return getIsRegexModifiedStore(n, result, factory);
        }
```

you can write

```
        if (ElementUtils.matchesElement(method, IS_REGEX_METHOD_NAME, String.class, int.class)
	    || ElementUtils.matchesElement(
                method, IS_REGEX_METHOD_NAME, String.class, int.class, int[].class)) {
	    ... code goes here ...
        }
```

which does not require a new method that moves the code far from where it is used.

The same comment applies to the next two tests.